src/idlcxx/src/streamer_generator.c

```diff
@@ -46,6 +46,14 @@ format_ostream_indented(indent ? ctx->depth*2 : 0, ctx->key_write_stream, _str,
 #define format_key_read_stream(indent,ctx,_str,...) \
 format_ostream_indented(indent ? ctx->depth*2 : 0, ctx->key_read_stream, _str, ##__VA_ARGS__);
 
+#define format_max_size_stream(indent,ctx,key,_str,...) \
+format_ostream_indented(indent ? ctx->depth*2 : 0, ctx->max_size_stream, _str, ##__VA_ARGS__); \
+if (key) { format_key_max_size_stream(indent, ctx, _str, ##__VA_ARGS__); }
+
+#define format_max_size_intermediate_stream(indent,ctx,key,_str,...) \
+format_ostream_indented(indent ? ctx->depth*2 : 0, ctx->max_size_intermediate_stream, _str, ##__VA_ARGS__); \
+if (key) { format_key_max_size_intermediate_stream(indent, ctx, _str, ##__VA_ARGS__); }
+
```

Comment:
Not a fan of using macro's like this (also not sure if the paste operator is used properly). You don't have to change it right now, but with constructions like this, I usually find there's a more elegant solution. If you must use macro's though, probably best to not use parameters twice (I could pass `some_char_pointer++`) and escape them to avoid the most obvious errors.

Reply:
Maybe a better approach is to not use macro's, but to define them as functions?
Though those functions would still require the re-use of variadic function parameters.